caddyhttp/httpserver/server.go

```diff
@@ -419,7 +419,7 @@ func (s *Server) serveHTTP(w http.ResponseWriter, r *http.Request) (int, error)
 
 	// we still check for ACME challenge if the vhost exists,
 	// because we must apply its HTTP challenge config settings
-	if caddytls.HTTPChallengeHandler(w, r, vhost.ListenHost) {
+	if caddytls.HTTPChallengeHandler(w, r, vhost.ListenHosts[0]) {
```

Comment:
How do we know the first ListenHost is the right one for the ACME challenge?

Reply:
We can't know. What I thought is, ListenHosts just behaves like the old ListenHost value. Maybe document is the right place to tell people that the first value will be used for ACME. This makes the implemetetion much simpler.